Authorization

dongiveajack opened the pull request:
added support for authorization

dongiveajack committed:
added authorization feature

dongiveajack commented on src/main/java/org/trips/service_framework/clients/RealmClient.java:
```diff
@@ -22,6 +23,9 @@
 @FeignClient(name = "realm", url = "${realm.base-url}")
 public interface RealmClient {
+    @RequestMapping(method = RequestMethod.POST, value = "/sessioninfo/authorize")
+    RealmSessionInfoResponse getSessionInfo(@RequestHeader("Cookie") String cookie, @RequestBody RealmAuthenticateRequest request);
```
will remove this redundant client call

dagaranupam commented on src/main/java/org/trips/service_framework/clients/RealmClient.java:
```diff
@@ -22,6 +23,9 @@
 @FeignClientAdvice(RealmExceptionHandler.class)
 @FeignClient(name = "realm", url = "${realm.base-url}")
 public interface RealmClient {
+    @RequestMapping(method = RequestMethod.POST, value = "/sessioninfo/authorize")
```
duplicate api definition. same as authorize present below

dongiveajack committed:
removed duplicate api call from realm client

dongiveajack committed:
Refactor request uri matching logic with AntPathMatcher

Replaced exact URI matching with pattern-based matching using AntPathMatcher. This allows for more flexible and efficient identification of allowed permissions based on API patterns. Added a new helper method to handle pattern matching logic.
